include/ddc/kernels/splines/bsplines_non_uniform.hpp

```diff
@@ -70,125 +95,302 @@ class NonUniformBSplines : detail::NonUniformBSplinesBase
+         *
+         * The provided break points describe the separation between the cells on which the polynomials
+         * comprising a spline are defined. They are used to build a set of knots. There are 2*degree more
+         * knots than break points. The knots are defined as follows:
+         * \f$ k_i = b_0 \forall 0 \leq i < d \f$
+         * \f$ k_{i+d} = b_i \forall 0 \leq i < n_b \f$
+         * \f$ k_{i+d+n_b} = b_{n_b} \forall 0 \leq i < d \f$
+         * where \f$d\f$ is the degree of the polynomials, and \f$n_b\f$ is the number of basis points.
+         *
+         * This constructor makes the knots accessible via a DiscreteSpace.
+         *
+         * @param breaks_begin The iterator which points at the beginning of the break points.
+         * @param breaks_end The iterator which points at the end of the break points.
+         */
         template <class RandomIt>
         Impl(RandomIt breaks_begin, RandomIt breaks_end);
 
+        /** @brief Copy-constructs from another Impl with a different Kokkos memory space
+         *
+         * @param impl A reference to the other Impl
+         */
+        template <class OriginMemorySpace>
+        explicit Impl(Impl<DDim, OriginMemorySpace> const& impl)
+            : m_domain(impl.m_domain)
+            , m_nknots(impl.m_nknots)
+        {
+        }
+
+        /** @brief Copy-constructs
+         *
+         * @param x A reference to another Impl
+         */
         Impl(Impl const& x) = default;
 
+        /** @brief Move-constructs
+         *
+         * @param x An rvalue to another Impl
+         */
         Impl(Impl&& x) = default;
 
+        /// @brief Destructs
         ~Impl() = default;
 
+        /** @brief Copy-assigns
+         *
+         * @param x A reference to another Impl
+         * @return A reference to the copied Impl
+         */
         Impl& operator=(Impl const& x) = default;
 
+        /** @brief Move-assigns
+         *
+         * @param x An rvalue to another Impl
+         * @return A reference to the moved Impl
+         */
         Impl& operator=(Impl&& x) = default;
 
+        /** @brief Evaluates non-zero B-splines at a given coordinate.
+         *
+         * The values are computed for every B-spline with support at the given coordinate x. There are only (degree+1)
+         * B-splines which are non-zero at any given point. It is these B-splines which are evaluated.
+         * This can be useful to calculate a spline approximation of a function. A spline approximation at coordinate x
+         * is a linear combination of these B-spline evaluations weighted with the spline coefficients of the spline-transformed
+         * initial discrete function.
+         *
+         * @param[out] values The values of the B-splines evaluated at coordinate x. It has to be a 1D mdspan with (degree+1) elements.
+         * @param[in] x The coordinate where B-splines are evaluated.
```

Comment:
Shall we mention the range of validity for `x` ? that is the range of break points right ?

Reply:
That is the correct range. I agree that this could be useful :+1: